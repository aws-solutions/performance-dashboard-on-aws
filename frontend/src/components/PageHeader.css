.Header {
    margin-top: 50px;
    margin-bottom: 65px;
}