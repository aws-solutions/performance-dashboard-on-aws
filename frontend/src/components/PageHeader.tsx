import React, { ReactNode } from 'react';
import './PageHeader.css';

interface Props {
    children: ReactNode,
}

function Subtitle(props: Props) {
    return (
        <p>{props.children}</p>
    );
}

function Title(props: Props) {
    return (
        <h1>{props.children}</h1>
    );
}

function PageHeader(props: Props) {
    return (
        <div className="Header">
            {props.children}
        </div>
    )
}

PageHeader.Title = Title;
PageHeader.Subtitle = Subtitle;

export default PageHeader;