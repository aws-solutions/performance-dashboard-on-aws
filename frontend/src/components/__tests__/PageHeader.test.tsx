import React from 'react';
import { render } from '@testing-library/react';
import PageHeader from '../PageHeader';

test('render a header with subtitle', () => {
  const { container } = render(
    <PageHeader>
        <PageHeader.Title>Some title</PageHeader.Title>
        <PageHeader.Subtitle>This is a subtitle</PageHeader.Subtitle>
    </PageHeader>
  );

  expect(container).toMatchSnapshot();
});
