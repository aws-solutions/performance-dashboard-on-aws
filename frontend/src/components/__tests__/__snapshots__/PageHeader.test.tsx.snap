// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a header with subtitle 1`] = `
<div>
  <div
    class="Header"
  >
    <h1>
      Some title
    </h1>
    <p>
      This is a subtitle
    </p>
  </div>
</div>
`;
